adaptors/proximityadaptor-evdev/proximityadaptor-evdev.cpp

```diff
@@ -57,27 +58,29 @@ void ProximityAdaptorEvdev::interpretEvent(int src, struct input_event *ev)
     Q_UNUSED(src);
 
     switch (ev->type) {
-        case SELFDEF_EV_SW:
+    case EV_ABS:
+    case SELFDEF_EV_SW:
```

Comment:
Even if this works, it is logically wrong. EV_ABS and EV_SW have different event codes and should not be handled in this way. If you want to use common code for value processing, then something like

    if( ev->type == EV_SW && ev->code == SW_FRONT_PROXIMITY ||
        ev->type == EV_ABS && ev->code == ABS_DISTANCE ) { check ev->value }